compiler/rustc_llvm/llvm-wrapper/RustWrapper.cpp

```diff
@@ -636,117 +635,63 @@ template <typename DIT> DIT *unwrapDIPtr(LLVMMetadataRef Ref) {
-  FlagBitField = (1 << 19),
-  FlagNoReturn = (1 << 20),
-  // Do not add values that are not supported by the minimum LLVM
-  // version we support! see llvm/include/llvm/IR/DebugInfoFlags.def
-};
-
-inline LLVMRustDIFlags operator&(LLVMRustDIFlags A, LLVMRustDIFlags B) {
-  return static_cast<LLVMRustDIFlags>(static_cast<uint32_t>(A) &
-                                      static_cast<uint32_t>(B));
-}
-
-inline LLVMRustDIFlags operator|(LLVMRustDIFlags A, LLVMRustDIFlags B) {
-  return static_cast<LLVMRustDIFlags>(static_cast<uint32_t>(A) |
-                                      static_cast<uint32_t>(B));
-}
-
-inline LLVMRustDIFlags &operator|=(LLVMRustDIFlags &A, LLVMRustDIFlags B) {
-  return A = A | B;
-}
-
-inline bool isSet(LLVMRustDIFlags F) { return F != LLVMRustDIFlags::FlagZero; }
-
-inline LLVMRustDIFlags visibility(LLVMRustDIFlags F) {
-  return static_cast<LLVMRustDIFlags>(static_cast<uint32_t>(F) & 0x3);
-}
-
-static DINode::DIFlags fromRust(LLVMRustDIFlags Flags) {
-  DINode::DIFlags Result = DINode::DIFlags::FlagZero;
-
-  switch (visibility(Flags)) {
-  case LLVMRustDIFlags::FlagPrivate:
-    Result |= DINode::DIFlags::FlagPrivate;
+static DINode::DIFlags fromRust(LLVMDIFlags Flags) {
+  using DIFlags = DINode::DIFlags;
+
+  // Internally, LLVM does this conversion with a simple integer cast, but we
+  // can't rely on that always being the case so we write this the long way.
+  auto Result = DIFlags::FlagZero;
+
+  // clang-format off
+  if (Flags & LLVMDIFlagFwdDecl)             Result |= DIFlags::FlagFwdDecl;
+  if (Flags & LLVMDIFlagAppleBlock)          Result |= DIFlags::FlagAppleBlock;
+  if (Flags & LLVMDIFlagReservedBit4)        Result |= DIFlags::FlagReservedBit4;
+  if (Flags & LLVMDIFlagVirtual)             Result |= DIFlags::FlagVirtual;
+  if (Flags & LLVMDIFlagArtificial)          Result |= DIFlags::FlagArtificial;
+  if (Flags & LLVMDIFlagExplicit)            Result |= DIFlags::FlagExplicit;
+  if (Flags & LLVMDIFlagPrototyped)          Result |= DIFlags::FlagPrototyped;
+  if (Flags & LLVMDIFlagObjcClassComplete)   Result |= DIFlags::FlagObjcClassComplete;
+  if (Flags & LLVMDIFlagObjectPointer)       Result |= DIFlags::FlagObjectPointer;
+  if (Flags & LLVMDIFlagVector)              Result |= DIFlags::FlagVector;
+  if (Flags & LLVMDIFlagStaticMember)        Result |= DIFlags::FlagStaticMember;
+  if (Flags & LLVMDIFlagLValueReference)     Result |= DIFlags::FlagLValueReference;
+  if (Flags & LLVMDIFlagRValueReference)     Result |= DIFlags::FlagRValueReference;
+  // This flag has been recycled, but the value in the C API hasn't been renamed yet.
+  if (Flags & LLVMDIFlagReserved)            Result |= DIFlags::FlagExportSymbols;
+  if (Flags & LLVMDIFlagSingleInheritance)   Result |= DIFlags::FlagSingleInheritance;
+  if (Flags & LLVMDIFlagMultipleInheritance) Result |= DIFlags::FlagMultipleInheritance;
+  if (Flags & LLVMDIFlagVirtualInheritance)  Result |= DIFlags::FlagVirtualInheritance;
```

Comment:
Actually, if I'm bothering to have special treatment for the `LLVMDIFlagAccessibility` flags below, it makes no sense to not also do the same thing for these `LLVMDIFlagPtrToMemberRep` flags as well.

(Other than the fact that Rust might never actually use these flags? But following that line of reasoning suggests perhaps not supporting conversion of them at all.)